db/write_callback_test.cc

```diff
@@ -184,7 +190,8 @@ TEST_F(WriteCallbackTest, WriteWithCallbackTest) {
                       ASSERT_TRUE(writer->callback->Callback(nullptr).ok() ==
                                   !write_group.back().callback_.should_fail_);
                     }
-
+                    
+                    threads_waiting.fetch_add(1);
                     // wait for friends
                     while (threads_waiting.load() < write_group.size()) {
```

Comment:
It is just waiting for other threads reach this point, and they are not doing a lot of work other than a few asserts in this callback. Busy waiting should be enough.